### PR TITLE
feat(server, autogpt): Add Example files and update build option

### PR DESCRIPTION
**Pull request by ntindle**
### Background

- We want to support one-click-ish install of autogpt and autogpt_server
- We need users to agree to terms to do so
- We need a way to host the frontend on the server

### Changes 🏗️

- Adds the example files
- updates cx_freeze
- update the server to serve the example files

To update the serving when needed, Swap example files directory to the correct build directory in setup.py in the include_files in the first tuple (source), and rename the second (dest) to something better.

Then go into data.py and update the path to the built frontend for local dev and the frozen version

### PR Quality Scorecard ✨

- [x] Have you used the PR description template? &ensp; `+2 pts`
- [ ] Is your pull request atomic, focusing on a single change? &ensp; `+5 pts`
- [ ] Have you linked the GitHub issue(s) that this PR addresses? &ensp; `+5 pts`
- [ ] Have you documented your changes clearly and comprehensively? &ensp; `+5 pts`
- [ ] Have you changed or added a feature? &ensp; `-4 pts`
  - [ ] Have you added/updated corresponding documentation? &ensp; `+4 pts`
  - [ ] Have you added/updated corresponding integration tests? &ensp; `+5 pts`
- [ ] Have you changed the behavior of AutoGPT? &ensp; `-5 pts`
  - [ ] Have you also run `agbenchmark` to verify that these changes do not regress performance? &ensp; `+10 pts`

___

### **PR Type**
Enhancement, Other

___

### **Description**
- Added a function to convert text files to RTF format in both `autogpt/setup.py` and `rnd/autogpt_server/setup.py`.
- Integrated the conversion of LICENSE to LICENSE.rtf in both setup files.
- Updated cx_Freeze setup configurations to include the RTF license file.
- Updated cx_Freeze dependency in `pyproject.toml` files to use a specific Git repository and branch in develop mode.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>setup.py</strong><dd><code>Add RTF conversion and update cx_Freeze setup</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
autogpt/setup.py

<li>Added a function to convert text files to RTF format.<br> <li> Integrated the conversion of LICENSE to LICENSE.rtf.<br> <li> Updated cx_Freeze setup to include the RTF license file.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7271/files#diff-3b73fb2db1dc7dab3c93a250a8495c74a46c417b43ca006fb1bf65c643aacad2">+38/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>setup.py</strong><dd><code>Add RTF conversion and update cx_Freeze setup for server</code>&nbsp; </dd></summary>
<hr>
      
rnd/autogpt_server/setup.py

<li>Added a function to convert text files to RTF format.<br> <li> Integrated the conversion of LICENSE to LICENSE.rtf.<br> <li> Updated cx_Freeze setup to include the RTF license file.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7271/files#diff-70ab2c5f550620f9edd017312b42ebe58b9fc3e858d84f759b4d1e08ddd3be6d">+39/-4</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Dependencies
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>pyproject.toml</strong><dd><code>Update cx_Freeze dependency to specific Git repository</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
autogpt/pyproject.toml

<li>Updated cx_Freeze dependency to use a specific Git repository and <br>branch in develop mode.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7271/files#diff-50997d0a38b1e338082b724342f48b3c32fae5b29a5391cbe7d7f9f0b1065dbc">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>pyproject.toml</strong><dd><code>Update cx_Freeze dependency to specific Git repository for server</code></dd></summary>
<hr>
      
rnd/autogpt_server/pyproject.toml

<li>Updated cx_Freeze dependency to use a specific Git repository and <br>branch in develop mode.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7271/files#diff-adb694b1ef2fb9dabbdd0efe486d08bc9336bcd7b824f13f4125228d70cd4d7a">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by ntindle** [comment]
The import works at runtime and I’m too tired to debug it. We need to run prisma generate and also probably create and push the SQLite db if it doesn’t exist just for stability. I can package in an empty one but I think it would be wise to create it for stability reasons

File: rnd/autogpt_server/autogpt_server/app.py [hunk 1]
```diff
@@ -19,10 +27,14 @@ def run_processes(processes: list[AppProcess], **kwargs):
 
 
 def main(**kwargs):
+    settings = get_config_and_secrets()
+    set_start_method("spawn", force=True)
+    freeze_support()
```
Comment (majdyz) [inline rnd/autogpt_server/autogpt_server/app.py:32]: These has been done on `process.py`

Comment (ntindle) [reply rnd/autogpt_server/autogpt_server/app.py:32]: i can do on both, but it was crashing when frozen if not here

Comment (ntindle) [reply rnd/autogpt_server/autogpt_server/app.py:32]: okay so if i set the spawn method here but not freeze, it works???

**Review by Swiftyos — APPROVED** [review]

**Comment by Pwuts** [comment]
This fails to install for me:
```
  - Updating cx-freeze (7.0.0 876fe77 -> 7.1.1 876fe77): Failed

  ChefBuildError

  Backend subprocess exited when trying to invoke build_editable
  
  running editable_wheel
  creating /tmp/poetry-chef-l92dv9z4/.tmp-zywo1jiy/cx_Freeze.egg-info
  writing /tmp/poetry-chef-l92dv9z4/.tmp-zywo1jiy/cx_Freeze.egg-info/PKG-INFO
  writing dependency_links to /tmp/poetry-chef-l92dv9z4/.tmp-zywo1jiy/cx_Freeze.egg-info/dependency_links.txt
  writing entry points to /tmp/poetry-chef-l92dv9z4/.tmp-zywo1jiy/cx_Freeze.egg-info/entry_points.txt
  writing requirements to /tmp/poetry-chef-l92dv9z4/.tmp-zywo1jiy/cx_Freeze.egg-info/requires.txt
  writing top-level names to /tmp/poetry-chef-l92dv9z4/.tmp-zywo1jiy/cx_Freeze.egg-info/top_level.txt
  writing manifest file '/tmp/poetry-chef-l92dv9z4/.tmp-zywo1jiy/cx_Freeze.egg-info/SOURCES.txt'
  reading manifest file '/tmp/poetry-chef-l92dv9z4/.tmp-zywo1jiy/cx_Freeze.egg-info/SOURCES.txt'
  reading manifest template 'MANIFEST.in'
  adding license file 'LICENSE.md'
  writing manifest file '/tmp/poetry-chef-l92dv9z4/.tmp-zywo1jiy/cx_Freeze.egg-info/SOURCES.txt'
  creating '/tmp/poetry-chef-l92dv9z4/.tmp-zywo1jiy/cx_Freeze-7.1.1.dist-info'
  creating /tmp/poetry-chef-l92dv9z4/.tmp-zywo1jiy/cx_Freeze-7.1.1.dist-info/WHEEL
  running build_py
  running build_ext
  creating /tmp/tmp28l8tueg.build-temp/source
  creating /tmp/tmp28l8tueg.build-temp/source/bases
  x86_64-linux-gnu-gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -fPIC -I/tmp/tmp5avh93ft/.venv/include -I/usr/include/python3.10 -c source/bases/console.c -o /tmp/tmp28l8tueg.build-temp/source/bases/console.o
  source/bases/console.c:7:10: fatal error: Python.h: No such file or directory
      7 | #include <Python.h>
        |          ^~~~~~~~~~
  compilation terminated.
  Traceback (most recent call last):
    File "/tmp/tmp5avh93ft/.venv/lib/python3.10/site-packages/setuptools/_distutils/spawn.py", line 71, in spawn
      subprocess.check_call(cmd, env=_inject_macos_ver(env))
    File "/usr/lib/python3.10/subprocess.py", line 369, in check_call
      raise CalledProcessError(retcode, cmd)
  subprocess.CalledProcessError: Command '['/usr/bin/x86_64-linux-gnu-gcc', '-Wno-unused-result', '-Wsign-compare', '-DNDEBUG', '-g', '-fwrapv', '-O2', '-Wall', '-g', '-fstack-protector-strong', '-Wformat', '-Werror=format-security', '-g', '-fwrapv', '-O2', '-fPIC', '-I/tmp/tmp5avh93ft/.venv/include', '-I/usr/include/python3.10', '-c', 'source/bases/console.c', '-o', '/tmp/tmp28l8tueg.build-temp/source/bases/console.o']' returned non-zero exit status 1.
  
  The above exception was the direct cause of the following exception:
  
  Traceback (most recent call last):
    File "/tmp/tmp5avh93ft/.venv/lib/python3.10/site-packages/setuptools/_distutils/unixccompiler.py", line 191, in _compile
      self.spawn(compiler_so + cc_args + [src, '-o', obj] + extra_postargs)
    File "/tmp/tmp5avh93ft/.venv/lib/python3.10/site-packages/setuptools/_distutils/ccompiler.py", line 1044, in spawn
      spawn(cmd, dry_run=self.dry_run, **kwargs)
    File "/tmp/tmp5avh93ft/.venv/lib/python3.10/site-packages/setuptools/_distutils/spawn.py", line 77, in spawn
      raise DistutilsExecError(
  distutils.errors.DistutilsExecError: command '/usr/bin/x86_64-linux-gnu-gcc' failed with exit code 1
  
  During handling of the above exception, another exception occurred:
  
  Traceback (most recent call last):
    File "/tmp/tmp5avh93ft/.venv/lib/python3.10/site-packages/setuptools/command/editable_wheel.py", line 153, in run
      self._create_wheel_file(bdist_wheel)
    File "/tmp/tmp5avh93ft/.venv/lib/python3.10/site-packages/setuptools/command/editable_wheel.py", line 355, in _create_wheel_file
      files, mapping = self._run_build_commands(dist_name, unpacked, lib, tmp)
    File "/tmp/tmp5avh93ft/.venv/lib/python3.10/site-packages/setuptools/command/editable_wheel.py", line 278, in _run_build_commands
      self._run_build_subcommands()
    File "/tmp/tmp5avh93ft/.venv/lib/python3.10/site-packages/setuptools/command/editable_wheel.py", line 305, in _run_build_subcommands
      self.run_command(name)
    File "/tmp/tmp5avh93ft/.venv/lib/python3.10/site-packages/setuptools/_distutils/cmd.py", line 316, in run_command
      self.distribution.run_command(command)
    File "/tmp/tmp5avh93ft/.venv/lib/python3.10/site-packages/setuptools/dist.py", line 974, in run_command
      super().run_command(command)
    File "/tmp/tmp5avh93ft/.venv/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 989, in run_command
      cmd_obj.run()
    File "<string>", line 261, in run
    File "/tmp/tmp5avh93ft/.venv/lib/python3.10/site-packages/setuptools/command/build_ext.py", line 93, in run
      _build_ext.run(self)
    File "/tmp/tmp5avh93ft/.venv/lib/python3.10/site-packages/setuptools/_distutils/command/build_ext.py", line 359, in run
      self.build_extensions()
    File "/tmp/tmp5avh93ft/.venv/lib/python3.10/site-packages/setuptools/_distutils/command/build_ext.py", line 479, in build_extensions
      self._build_extensions_serial()
    File "/tmp/tmp5avh93ft/.venv/lib/python3.10/site-packages/setuptools/_distutils/command/build_ext.py", line 505, in _build_extensions_serial
      self.build_extension(ext)
    File "<string>", line 46, in build_extension
    File "/tmp/tmp5avh93ft/.venv/lib/python3.10/site-packages/setuptools/_distutils/ccompiler.py", line 605, in compile
      self._compile(obj, src, ext, cc_args, extra_postargs, pp_opts)
    File "/tmp/tmp5avh93ft/.venv/lib/python3.10/site-packages/setuptools/_distutils/unixccompiler.py", line 193, in _compile
      raise CompileError(msg)
  distutils.errors.CompileError: command '/usr/bin/x86_64-linux-gnu-gcc' failed with exit code 1
  /tmp/tmp5avh93ft/.venv/lib/python3.10/site-packages/setuptools/_distutils/dist.py:989: _DebuggingTips: Problem in editable installation.
  !!
  
          ********************************************************************************
          An error happened while installing `cx_Freeze` in editable mode.
  
          The following steps are recommended to help debug this problem:
  
          - Try to install the project normally, without using the editable mode.
            Does the error still persist?
            (If it does, try fixing the problem before attempting the editable mode).
          - If you are using binary extensions, make sure you have all OS-level
            dependencies installed (e.g. compilers, toolchains, binary libraries, ...).
          - Try the latest version of setuptools (maybe the error was already fixed).
          - If you (or your project dependencies) are using any setuptools extension
            or customization, make sure they support the editable mode.
  
          After following the steps above, if the problem still persists and
          you think this is related to how setuptools handles editable installations,
          please submit a reproducible example
          (see https://stackoverflow.com/help/minimal-reproducible-example) to:
  
              https://github.com/pypa/setuptools/issues
  
          See https://setuptools.pypa.io/en/latest/userguide/development_mode.html for details.
          ********************************************************************************
  
  !!
    cmd_obj.run()
  error: command '/usr/bin/x86_64-linux-gnu-gcc' failed with exit code 1
```

**Pull request merged**